erigon-lib/commitment/hex_patricia_hashed.go

```diff
@@ -2033,31 +2064,46 @@ func (hph *HexPatriciaHashed) Process(ctx context.Context, updates *Updates, log
 
 		default:
 		}
-
+		hph.metrics.Accounts.UpdatesInc(plainKey)
+		hph.metrics.Accounts.currentPlainKey = plainKey
+		hph.metrics.Accounts.UpdatesStorageInc(plainKey)
 		if hph.trace {
 			fmt.Printf("\n%d/%d) plainKey [%x] hashedKey [%x] currentKey [%x]\n", ki+1, updatesCount, plainKey, hashedKey, hph.currentKey[:hph.currentKeyLen])
 		}
 		// Keep folding until the currentKey is the prefix of the key we modify
 		for hph.needFolding(hashedKey) {
+			startFold := hph.metrics.Now()
+			hph.metrics.Accounts.FoldsInc(hph.metrics.Accounts.currentPlainKey)
 			if err := hph.fold(); err != nil {
 				return fmt.Errorf("fold: %w", err)
 			}
+			hph.metrics.TotalFoldingTimeInc(startFold)
+			hph.metrics.Accounts.TotalFoldingTimeInc(hph.metrics.Accounts.currentPlainKey, startFold)
 		}
 		// Now unfold until we step on an empty cell
 		for unfolding := hph.needUnfolding(hashedKey); unfolding > 0; unfolding = hph.needUnfolding(hashedKey) {
+			startUnfold := hph.metrics.Now()
+			hph.metrics.Accounts.UnfoldsInc(hph.metrics.Accounts.currentPlainKey)
+			hph.metrics.Unfolds.Add(1)
 			if err := hph.unfold(hashedKey, unfolding); err != nil {
 				return fmt.Errorf("unfold: %w", err)
 			}
+			hph.metrics.TotalUnfoldingTimeInc(startUnfold)
+			hph.metrics.Accounts.TotalUnfoldingTimeInc(hph.metrics.Accounts.currentPlainKey, startUnfold)
 		}
 
 		if stateUpdate == nil {
 			// Update the cell
 			if len(plainKey) == hph.accountKeyLen {
+				hph.metrics.LoadAccount.Add(1)
```

Comment:
here as well why make 2 calls when can do one function controlling this two changes